en/asgardeo/docs/complete-guides/javascript/manage-tokens-in-apps.md

```diff
@@ -0,0 +1,122 @@
+
+- **Asynchronous** Task Management: Web workers enhance the handling of asynchronous tasks, such as token renewal or data fetching, without blocking the main thread.
+
+- **Scalability:** By enabling parallelism, web workers make applications more responsive and better equipped to handle multiple concurrent operations.
+
+## Initiate Logout
+
+The Asgardeo JavaScript SDK provides a simple approach  to handle user logout from your app. When a user logs out, the SDK ensures that both the local session and the session on the {{product_name}} are terminated, ensuring a complete and secure logout process and you don’t need to worry on cleanup activities
+
+When a user initiates log out, the following steps typically occur.
+
+- **Local Session Termination:** The SDK clears any locally stored credentials, such as the access token and the ID token, which are used to maintain the user's authentication state within the application. This effectively logs the user out of the application locally.
+
+- **Redirection to {{product_name}} for sign out:** After clearing the local session, the SDK redirects the user to the sign-out endpoint of your {{product_name}} organization. This ensures that the user is also signed out globally from {{product_name}}. It’s particularly important in single sign-on (SSO) scenarios where the user might be logged into multiple applications under the same identity.
+
+- **Post Sign-Out Redirection:** Once the global sign-out is complete, the user is redirected back to a specified URL, usually the application's homepage or a custom logout page, which is configured in the SDK's authConfig under signOutRedirectURL.
+
+## Silent Sign In
+
+Silent login allows an app to check if a user is already authenticated, either through a session cookie or a token stored in the browser, and re-authenticate automatically in the background. To implement silent sign-in using the Asgardeo JavaScript SDK, you can leverage the library's built-in support for token renewal and session management. Here's how you can do it:
+
+- **Configure the Silent Sign-In:** Ensure that your `authConfig` is set up to allow silent sign-in. You need to configure the prompt parameter to `none` when initiating a silent sign-in request. This instructs the identity provider to not display any login prompts and to rely on existing sessions instead.
+
+- **Use the SDK’s Built-in Functionality:** The Asgardeo JavaScript SDK typically handles silent token renewal automatically if the configuration is set correctly. When the access token is about to expire, the SDK will attempt to renew it silently in the background.
+
+- **Handling Token Expiry:** In your JavaScript components, you can handle token expiry by checking the authentication state and initiating a silent sign-in if the user’s session is still valid but the token has expired.
+
+```javascript
```

Comment:
Shall we update the formatting like this?

```js
(async () => {
  let user = undefined;

  // If there are auth search params i.e code, session_state, automatically trigger login.
  // Else, try to see if there's a session.
  if (AsgardeoAuth.SPAUtils.hasAuthSearchParamsInURL()) {
    user = await auth.signIn({ callOnlyOnRedirect: true });
  } else {
    user = await auth.trySignInSilently();
  }

  // Update the UI accordingly.
  if (user) {
    document.getElementById('authenticated-view').style.display = 'block';
    document.getElementById('unauthenticated-view').style.display = 'none';
    document.getElementById('username').innerHTML = user.username;
  } else {
    document.getElementById('authenticated-view').style.display = 'none';
    document.getElementById('unauthenticated-view').style.display = 'block';
  }
})();
```

Reply:
Done